Added Map Tier Functionality for Global KZ Maps

teammate165 opened the pull request:
Requires SteamWorks and SMJansson include to compile and their extension files to function.

Default value for Max Tier changed to 7
Generalized Chat Prefixes
Added missing includes for compiling.

teammate165 committed:
Added Map Tier Functionality for Global KZ Maps

Requires SteamWorks and SMJansson include to compile and their extension files to function.
Default value for Max Tier changed to 7
Generalized Chat Prefixes

Added missing includes for compiling.

sneak-it commented on addons/sourcemod/scripting/nominations_extended.sp:
```diff
@@ -189,6 +206,18 @@ public void OnSettingsChanged(Handle convar, const char[] oldValue, const char[]
 	}
 }
 
+public void OnMapStart()
```
This should happen only if KZTimer or GOKZ are found, and if globals are enabled

sneak-it replied:
Follow-up: In its current state, this will trigger on every server that runs this plugin, KZ or not, which is unnecessary. This will also run on non-global servers as well. If you would like it to run on non-global servers, perhaps add a convar switch, but generally speaking if it is non-global there is a reason for it.

sneak-it commented on addons/sourcemod/scripting/nominations_extended.sp:
```diff
@@ -134,6 +143,10 @@ public void OnLibraryAdded(const char[] szName)
 	}
+	if (StrEqual (szName, "gokz-core"))
+	{
+		g_bKzTimer = true;
```
GOKZ should be on its own boolean - returning that KZTimer exists breaks GOKZ usability with other parts of this plugin.

sneak-it commented on addons/sourcemod/scripting/nominations_extended.sp:
```diff
@@ -150,6 +163,10 @@ public void OnLibraryRemoved(const char[] szName)
 	}
+	if (StrEqual (szName, "gokz-core"))
+	{
+		g_bKzTimer = false;
```
ditto with above

sneak-it commented on addons/sourcemod/scripting/nominations_extended.sp:
```diff
@@ -225,7 +254,7 @@ public Action Command_Addmap(int client, int args)
 	if (args < 1)
 	{
-		CReplyToCommand(client, "[SNK.SRV] Usage: sm_nominate_addmap <mapname>");
+		CReplyToCommand(client, "[NE] Usage: sm_nominate_addmap <mapname>");
```
This prefix should be replaced with `g_szChatPrefix` and preferably stored as a translation

sneak-it reviewed (changes requested):
See my notes for requested changes, also:
- Please remove additional includes that are not shipped with MCE - we do not want to be responsible for keeping includes up to date. These would go great as Compilation Requirements linked in a Readme file (steamworks, colors, emitsoundany, smjansson)
- I generally try to stay away from 3rd party extension requirements due to them being an absolute pain to recompile when they break after major game/SM updates. Because global API plugins use it I won't require it for now, but if you could utilize the sm-json library which takes care of json responses rather than SMJansson that would be a huge plus: https://github.com/clugg/sm-json

Thank you!

teammate165 committed:
Replaced GOKZ, KZTimer check with GlobalAPI plugin check

A replacement for the double Library check to account for both GOKZ/KZTimer.

However I haven't tested this on a LAN server where people usually ignore the GlobalAPI error (usually due to missing SMJansson/Steamworks extensiosn). I'm hoping that since the plugin failed to load it's library won't be registered hence the bool value for KzGlobal will be false.

Minor changes:
- Proper usage of ChatPrefix Convar
- Commented the "_" line in universal.txt
- Renaming the rockthevote_extended config file, enforcing a naming scheme
- Removed 3rd party includes
- Added guide for compiling.

teammate165 committed:
Added a Global check for sending requests to GlobalAPI

oopsie daisy

teammate165 commented:
Apologies for disappearing, now that I have alot of free time I'd like to continue working on this.

Instead of checking for KZTimer and GOKZ libraries, we'll check for GlobalAPI's library. However I'm unsure whether GlobalAPI's library will still be registered if the plugin fails to load (most common reason for failure would be missing SMJansson/Steamworks). I'm thinking if the plugin fails to load due to missing dependencies, some of its function such as RegPluginLibrary won't be executed. Please correct me if I'm wrong.

As for integrating sm-json, I'll try to look up examples of its used and hopefully get it working in MCE.
